clients/java/src/main/java/io/camunda/zeebe/client/impl/http/ApiEntityConsumer.java

```diff
@@ -87,13 +92,16 @@ protected ApiEntity<T> generateContent() throws IOException {
       if (nonJsonBody == null || bufferedBytes == 0) {
         return null;
       }
+      if (responseType == ResponseType.TEXT) {
+        return (ApiEntity<T>) ApiEntity.of(new String(nonJsonBody, 0, bufferedBytes));
+      }
 
       return ApiEntity.of(ByteBuffer.wrap(nonJsonBody, 0, bufferedBytes));
```

Comment:
🔧 The `ResponseType` could have a `wrapValue` method or similar that returns the wrapped `nonJsonBody` when it's handed in. That way, we don't have to `if-return` here again, we already have a type and it should know how to return the wrapped value.

💭 It might actually make more sense to create a response object that encapsulates the logic of the underlying content/response type. Otherwise, this method here will become hard to understand really quickly if we add more supported response types. For example, the response object can hold a parser and buffer if it's a JSON response (JSON or JSON_PROBLEM). For XML text, the response object wouldn't have such objects at all. I can also create a commit to visualize what I have in mind later, if you like.